src/main/java/frc/robot/poseestimator/helpers/ObservationAccumulator.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot.poseestimator.helpers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class ObservationAccumulator<T> {
+
+	private final Supplier<List<T>> observationsSupplier;
+	private final int maxDataIntakes;
+	private List<T> accumulatedObservations;
+	private int dataIntakeCounter;
+
+	public ObservationAccumulator(Supplier<List<T>> observationSupplier, int maxCount) {
+		this .observationsSupplier = observationSupplier;
+		this.maxDataIntakes = maxCount;
+		this .accumulatedObservations = new ArrayList<>();
+		this .dataIntakeCounter = 0;
+	}
+
+	public List<T> getDataIntakeList() {
```

Comment:
You'll try to be as generic as you can when using requiring a specific type, but as specific as you can when giving a specific type. Maybe the user would want to use `ArrayList` features (and if he'll require `List`, polymorphism will do its thing).